### ISSUE TITLE
[list] keyboard navigation has extra stops horizontally

### ISSUE THREAD

**Issue by paulcpederson** [Esri/calcite-components#12856]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

The left and right arrows appear to support focusing on two separate tab stops inside a list component even with no interactive elements slotted in. Is this intentional?

<img width="497" height="210" alt="Image" src="https://github.com/user-attachments/assets/4c212c40-8f05-495e-8b71-b0088bcca127" />
<img width="497" height="283" alt="Image" src="https://github.com/user-attachments/assets/1dbc9651-0852-496f-afb5-38a7e759d917" />

### Actual Behavior

When you use keyboard arrows to navigate, you are able to set focus twice per row. This has been flagged by a PE on our team as an accessibility violation. 

### Expected Behavior

In a list item with no slotted actions or other focusable elements, only one column should be present

### Reproduction Sample

https://codepen.io/paulcp/pen/GgpaJVO

### Reproduction Steps

1. Click on a list item to focus it
2. Hit the right arrow, notice the slightly different focus outline.

### Reproduction Version

3.2

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Online
